[ML] add new `check_frequency` to delayed data check configuration in datafeeds (#76187)
elastic/elasticsearch: docs/reference/ml/ml-shared.asciidoc

This check runs only on real-time {dfeeds}.
end::delayed-data-check-config[]

tag::delayed-data-check-config-check-frequency[]

> The how often should the {dfeed} search for late data.

Reviewer: ```suggestion
Specifies how often the {dfeed} searches for late data.
```

It must be smaller than the delayed data `check_window`.
Additionally, if {dfeed} frequency is set, `check_frequency` must
be larger than `frequency`.

> Delayed data checks cannot occur more often than the configured `frequency`.

Reviewer: I think this sentence has the same meaning as the previous sentence
```suggestion

```

> But, setting `check_frequency` higher will increase the time between delayed

Reviewer: If I'm interpreting it correctly, this seems like a "no duh" sentence, so I'd suggest omitting this one too:
> But, setting `check_frequency` higher will increase the time between delayed

data checks.

> 

Reviewer: This final paragraph is not being formatted properly in the output. It either needs to be linked to the previous paragraph with a "+" or else just merged into the previous paragraph (I think the latter's my preference)
```suggestion
+
```
